src/types.ts

```diff
@@ -0,0 +1,132 @@
+import { KintoIdObject, Permission } from "kinto-http";
+import Collection from "./collection";
+
+export type $TSFixMe = any;
+
+export type WithOptional<T, K extends keyof T> = Omit<T, K> &
+  Partial<Pick<T, K>>;
+
+export type WithRequired<T, K extends keyof T> = T & Required<Pick<T, K>>;
+
+export interface IdSchema {
+  generate(record?: any): string;
+  validate(id: string): boolean;
+}
+
+export interface RemoteTransformer {
+  encode(record: any): any;
+  decode(record: any): any;
+}
+
+export type AvailableHook = "incoming-changes";
+
+export type Hooks<
+  T extends { id: string; [key: string]: unknown } = { id: string }
+> = {
+  [key in AvailableHook]?: ((
+    record: any,
+    collection: Collection<T>
+  ) => Promise<{ changes: T[] }>)[];
+};
+
+export interface KintoRepresentation<T = unknown> {
+  data: T;
+  permissions: { [key in Permission]?: string[] };
+}
```

Comment:
Don't we want to reuse kinto-http.js types for this?

Reply:
I'll take another look at where we're using `KintoRepresentation`, but technically the equivalent in `kinto-http.js` for this type has some additional required fields on the `data` property that I think we were missing in this library.